src/button-group/button-group.js

```diff
@@ -49,12 +51,18 @@ function getBorderRadii(index, length) {
   };
 }
 
-export default function ButtonGroup(props: PropsT) {
+type LocaleT = {|locale?: ButtonGroupLocaleT|};
+export function ButtonGroupRoot(props: {|...PropsT, ...LocaleT|}) {
   const {overrides = {}} = props;
   const [Root, rootProps] = getOverrides(overrides.Root, StyledRoot);
 
   return (
-    <Root aria-label={props.ariaLabel} {...rootProps}>
+    <Root
```

Comment:
could context consumer be placed here rather than in the wrapper component? not sure why locale needs to be provided through props

Reply:
yeah, sadly enyzme does not fully support contexts, so sadly the test framework affected our production code :( 

is this correct @tajo ?

Reply:
Yea, this is a workaround for Enzyme. [It can't](https://github.com/airbnb/enzyme/issues/1908#issuecomment-439747826) shallow render the `renderProp`.

Reply:
could we simply `mount` and avoid the indirection? i've found that code bases grow in puzzling ways when ad hoc refactoring to enable test patterns

Reply:
I agree, 100%.

We've tried that too, it won't work - I think Vojtech will want to bring up the issues with enzyme, and start a chat on what we can do about it

for now, would you be ok with it as is, and fix it once we figure out what to do with enzyme?

Reply:
for sure - i would personally prefer to remove the test than add the additional indirection, but up to you